paper/paper.md

```diff
@@ -0,0 +1,111 @@
+
+TODO WRITING: insert fig?
+
+#### Using Gecko Annotations
+
+To improve model accuracy and improve coverage of edge cases, users may find it useful to use Gecko: a free web application for manual segmentation of an audio files by speaker as well as annotation of the linguistic content of a conversation [@Gecko2019]. Speakerbox makes use of Gecko annotations as a method for training dataset creation by providing functions to split and prepare audio files using the annotations stored in a Gecko created JSON file.
+
+### Data Preparation
+
+Speakerbox is built with the goal of making multi-recording, multi-speaker identification model training as simple as possible while still attempting to help train an accurate, useful model for application. To ensure that the model is learning the features of each speaker's voice and not the features of the microphone or the specific words and phrases of the recording, we create dataset training, test, and evaluation splits based off of a recording holdout and speaker stratification pattern. Each train, test, or evaluation subset must contain unique recording IDs to reduce the chance of learning the features of specific microphones or recording contexts. We then check that our speaker stratifaction requirement is met by checking that each of the produced subsets contains recordings of every speaker available from the whole dataset. For example, if there are nine unique speakers in the dataset as a whole, then each train, test, and evaluation subset is required to have all nine speakers as well.
+
+If either of these conditions are not met, Speakerbox retries this random sampling process. If we cannot find a valid recording holdout and speaker stratification configuration given the sampling iterations, we inform the user of this failure and ask them to add more examples to the dataset.
+
+TODO WRITING -- INSERT FIG OF THIS?
+
+TODO CODING -- allow param of "no holdout" and "no stratify" to ignore this process.
+
+### Model Training
+
+The Speakerbox training process consists of fine-tuning a pre-trained Wav2Vec2 speaker identification model [@yang2021superb] provided by Huggingface's Transformers library [@wolf-etal-2020-transformers]. The default model for fine-tuning ([superb/wav2vec2-base-superb-sid](https://huggingface.co/superb/wav2vec2-base-superb-sid)) was pre-trained on the VoxCeleb1 dataset [@Nagrani17].
+
+## Model Application
+
+TODO CODING -- WRITE APPLICATION FUNCTION / MOVE IT OVER TO THIS REPO FROM CDP-BACKEND AND ALSO SPLIT DEPS TO MAKE TRAINING AND APPLICATION DEPS DIFFERENT.
+
+## Usage in Existing Research
+
+We are utilizing Speakerbox trained models to annotate municipal council meeting transcripts provided by the Council Data Project [@Brown2021].
+
+Our end goals of this work are:
```

Comment:
I think instead of saying what the goals are - talk about the outcomes of using SB in a specific case